packages/react-native-platform-override/src/Manifest.ts

```diff
@@ -5,183 +5,91 @@
-   * should be present and unchanged since entry creation.
-   */
-  async validate(): Promise<Array<ValidationError>> {
-    const errors: Array<ValidationError> = [];
-
-    const manifestFiles = this.overrides.map(override => override.file);
-    const overrideFiles = await this.overrideRepo.listFiles();
-
-    const fileMissingFromManifest = _.difference(overrideFiles, manifestFiles);
-    fileMissingFromManifest.forEach(file =>
-      errors.push({type: 'fileMissingFromManifest', file: file}),
-    );
-    const overridesNotFound = _.difference(manifestFiles, overrideFiles);
-    overridesNotFound.forEach(file =>
-      errors.push({type: 'overrideFileNotFound', file: file}),
-    );
-
-    await Promise.all(
-      this.overrides.map(async override => {
-        if (override.type === 'platform') {
-          return;
-        }
-
-        const baseFile = override.baseFile;
-        const baseContent = await this.reactRepo.getFileContents(baseFile);
-        if (baseContent === null) {
-          errors.push({type: 'baseFileNotFound', file: override.file});
-          return;
-        }
-
-        const baseHash = Manifest.hashContent(baseContent);
-        if (baseHash.toLowerCase() !== override.baseHash.toLowerCase()) {
-          errors.push({type: 'outOfDate', file: override.file});
-          return;
-        }
-      }),
-    );
-
-    return errors;
+  static fromSerialized(man: Serialized.Manifest): Manifest {
+    const overrides = man.overrides.map(deserializeOverride);
+    return new Manifest(overrides);
   }
 
   /**
-   * Add an override to the manifest, throwing if any part is invalid.
+   * Check that overrides are accurately accounted for in the manifest. E.g.
+   * all files should be accounted for, and base files should be up to date
+   * with upstream.
    */
-  async addOverride(
-    type: OverrideType,
-    file: string,
-    baseFile?: string,
-    issue?: number,
-  ) {
-    if (['derived', 'patch', 'copy'].includes(type) && !baseFile) {
-      throw new Error(`baseFile is required for ${type} overrides`);
-    }
-
-    if (['patch', 'copy'].includes(type) && !issue) {
-      throw new Error(`issue is required for ${type} overrides`);
-    }
-
-    switch (type) {
-      case 'platform':
-        return this.addPlatformOverride(file);
-
-      case 'derived':
-        return this.addDerivedOverride(file, baseFile, issue);
-
-      case 'patch':
-        return this.addPatchOverride(file, baseFile, issue);
+  async validate(
+    overrideRepo: OverrideFileRepository,
+    reactRepo: ReactFileRepository,
+  ): Promise<Array<ValidationError>> {
+    const errors: ValidationError[] = [];
 
-      case 'copy':
-        return this.addCopyOverride(file, baseFile, issue);
+    const overrideFiles = await overrideRepo.listFiles();
+    const missingFromManifest = overrideFiles.filter(
+      file => !this.overrides.some(manOvr => manOvr.includesFile(file)),
```

Comment:
Just curious: Any particular reason why using `some` instead of `findOverrideIndex` in all the other places...

Reply:
Here we want to check if an override contains a file instead of the override name. These are effectively the same now but won't be with directory based overrides.